README.md

```diff
@@ -327,6 +338,185 @@ Default: `{{ __mssql_client_repository }}`
 
 Type: `string`
 
+### `mssql_ha_configure`
+
+Use the variables with the 'mssql_ha_' prefix to configure an SQL Server Always
+On availability group to provide high availability.
+
+Configuring for high availability is not supported on RHEL 7 because the
+System Roles ha_cluster role does not support RHEL 7.
+
+Set to `true` to configure for high availability. Setting to `false` does not
+remove configuration for high availability.
+
+When set to `true`, the role performs the following tasks:
+1. Include the System Roles firewall role to configure firewall:
+1.1. Open the firewall port set with the `mssql_ha_listener_port` variable.
+1.2. Enable the `high-availability` service in firewall.
+2. Configure SQL Server for high availability:
+2.1. Enable AlwaysOn Health events.
+2.2. Create certificate on the primary replica and distribute to other replicas.
+2.3. Configure endpoint and availability group.
+2.4. Configure the user provided with the ``mssql_ha_login` variable for
+     Pacemaker.
+3. Include the System Roles ha_cluster role to configure pacemaker.
+   Note that this role does not configure STONITH devices in Pacemaker.
```

Comment:
My intent here was to avoid making the mssql role wrap around the ha_cluster role much. We are going to provide example playbooks that will run the mssql role and then run the ha_cluster role with all requried ha_cluster-related variables.
I thought that setting `ha_cluster_resource_primitives` is not required because `stonith-enabled` is set to `true` by default, and it seemed to work. Could you please provide an example of what `mssql_ha_stonith_resources` value should look like?
In the mssql role, I guess adding `mssql_ha_stonith_resources` would look like this?
```
        ha_cluster_resource_primitives:
          - "{{ mssql_ha_stonith_resources }}"
          - id: ag_cluster
          ...
```

Reply:
> We are going to provide example playbooks that will run the mssql role and then run the ha_cluster role with all requried ha_cluster-related variables.

Ok, that would work for me. Just bear in mind this part of the ha_cluster role readme: "The role replaces the configuration of HA Cluster on specified nodes. Any settings not specified in the role variables will be lost."

> I thought that setting `ha_cluster_resource_primitives` is not required because `stonith-enabled` is set to `true` by default

That merely enables stonith. You still need to tell the cluster which stonith devices it can use, how it can connect to them, and which nodes they can fence. That is done by creating stonith resources. For further details, see for example:
* https://clusterlabs.org/pacemaker/doc/2.1/Clusters_from_Scratch/html/fencing.html
* https://clusterlabs.org/pacemaker/doc/2.1/Pacemaker_Explained/html/fencing.html
* https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_high_availability_clusters/assembly_configuring-fencing-configuring-and-managing-high-availability-clusters

> Could you please provide an example of what mssql_ha_stonith_resources value should look like?
```yaml
ha_cluster_resource_primitives:
  - id: myapc
    agent: stonith:fence_apc_snmp
    instance_attrs:
      - attrs:
          - name: ipaddr
            value: apc-switch.example.com
          - name: pcmk_host_map
            value: rhel8-node1.example.com:1;rhel8-node2.example.com:2
          - name: login
            value: apc
          - name: passwd
            value: apc
```

Reply:
> Ok, that would work for me. Just bear in mind this part of the ha_cluster role readme: "The role replaces the configuration > of HA Cluster on specified nodes. Any settings not specified in the role variables will be lost."

Oh, so in the case where a user runs the mssql role and then runs the ha_cluster role with `ha_cluster_resource_primitives` containing one item, the configuration applied by the mssql role would be removed? In this case, the mssql role would need to completely wrap around the ha_cluster role. As well as any other role that would use ha_cluster.

Reply:
@tomjelinek, in the case where a user runs the `mssql` role and then runs the `ha_cluster` role with `ha_cluster_resource_primitives` containing one item, the cluster configuration applied previously by the `mssql` role would be removed?

Reply:
> @tomjelinek, in the case where a user runs the `mssql` role and then runs the `ha_cluster` role with `ha_cluster_resource_primitives` containing one item, the cluster configuration applied previously by the `mssql` role would be removed?

In the case where a user runs the `mssql` role and then runs the `ha_cluster` role (with any settings), the cluster configuration applied previously by the `mssql` role would be removed.

It is unfortunate, but it is the best I can do now. Making the `ha_cluster` role work in a way where users can specify only partial cluster configuration and the role would not touch other parts of the cluster requires a lot more resources.

Reply:
> It is unfortunate, but it is the best I can do now. Making the ha_cluster role work in a way where users can specify only partial cluster configuration and the role would not touch other parts of the cluster requires a lot more resources.

I think this is fine.  We just document that you cannot use the ha_cluster role on mssql nodes, and vice versa.